internal/sourcemap/mapper.go

```diff
@@ -0,0 +1,270 @@
+package sourcemap
+
+import (
+	"sort"
+
+	"github.com/go-json-experiment/json"
+	"github.com/microsoft/typescript-go/internal/core"
+	"github.com/microsoft/typescript-go/internal/tspath"
+)
+
+// DocumentPosition represents a position in a document
+type DocumentPosition struct {
+	FileName string
+	Pos      int
+}
+
+// DocumentPositionMapper maps positions between source and generated files
+type DocumentPositionMapper interface {
+	GetSourcePosition(input DocumentPosition) DocumentPosition
+	GetGeneratedPosition(input DocumentPosition) DocumentPosition
+}
+
+// DocumentPositionMapperHost provides file system access for position mapping
+type DocumentPositionMapperHost interface {
+	GetSourceFileLike(fileName string) SourceFileLike
+	GetCanonicalFileName(path string) string
+	Log(text string)
+}
+
+// SourceFileLike represents a file with text content and line information
+type SourceFileLike interface {
+	Text() string
+	LineStarts() []core.TextPos
+}
+
+
+// MappedPosition represents a position mapping with additional metadata
+type MappedPosition struct {
+	GeneratedPosition int
+	SourceIndex       int
+	SourcePosition    int
+	SourceFileName    string // Resolved source file name
+}
+
+// CreateDocumentPositionMapper creates a document position mapper from a source map
+func CreateDocumentPositionMapper(host DocumentPositionMapperHost, mapContent string, mapPath string) DocumentPositionMapper {
+	var sourceMap RawSourceMap
+	if err := json.Unmarshal([]byte(mapContent), &sourceMap); err != nil {
+		host.Log("Failed to parse source map: " + err.Error())
```

Comment:
The old code didn't do the logging, did we handle this previously? I don't think adding logging to the source map package is good in general, but I could be wrong.